proto/p4/p4runtime.proto

```diff
@@ -356,6 +357,29 @@ message CloneSessionEntry {
   uint64 packet_length_bytes = 4;
 }
 
+//------------------------------------------------------------------------------
+// For writing and reading matches in a parser value set. A state transition
+// on an empty value set will never be taken. Matches contained in a value set
+// must be unique. The number of matches must be at most the size of the value
```

Comment:
Regarding uniqueness: Is the expectation that P4Runtime server will validate and flag this by returning an error? (I hope so). If yes, suggest clarifying this (I propose INVALID_ARGUMENT as the code returned for such an update in the WriteRequest).

Reply:
Why the requirement for uniqueness?  I can't think of anything that would go wrong in an implementation if it had overlapping ranges, or multiple value/mask entries that both matched the same value.

Reply:
I like having the uniqueness requirement better. Requiring entries to be unique doesn't mean disallowing overlap between matches. It means rejecting multiple occurrences of the exact same value, just like we did for table entries.

Reply:
Additionally if we don't add a priority value like we have for table entries, we should probably explicitly call out in the comment that the order of entries in the repeated field determines the priority (with the first value having the highest priority).

Reply:
It seems possible to define the same LHS value in the transition select cases for different transitions. So uniqueness is not required. It is kind of trick to implement too. But I do agree with clarifying the priority. Done.